pkg/router/router.go

```diff
@@ -2,8 +2,11 @@ package router
 
 import flaggerv1 "github.com/weaveworks/flagger/pkg/apis/flagger/v1beta1"
 
+const configAnnotation = "flagger.kubernetes.io/original-configuration"
```

Comment:
```suggestion
const configAnnotation = "finalizer.flagger.app/original-configuration"
```

Could we use `kubectl.kubernetes.io/last-applied-configuration` instead of duplicating the object serialization? As long the as the original version was applied with kubectl should work.

Reply:
I like the idea, it solves for kubectl and kustomize.  You have any issues with using the flagger annotation in the event `last-applied-configuration` is not present?  This would solve for helm and eliminate duplicated object serialization

Reply:
Wouldn't Helm remove the Flagger annotation on every upgrade and undo the routing? I think that if you have the virtual service in a chart, this would cause downtime for the app on every upgrade: Helm will change the selector, traffic will fail since the original deployment is scaled to zero, Flagger will notice the drift at some point and restore the virtual service.

Reply:
From initial testing helm wouldn't notice the annotation since the annotation was added outside the scope of a release.  Ultimately, helm's source of truth isn't the cluster it is either a configMap or secret depending on helm version or designation of release storage type.  I would expect (and I will do additional testing), if the VS spec was mutated by Flagger then Helm will not see those changes.  It will evaluate the configMap/Secret and see no change in the release.  I have tested this pretty heavily with Helm 2, but I need to continue my testing with Helm 3.